sources/packages/web/src/views/student/AppStudent.vue

```diff
@@ -12,32 +12,43 @@ import { AppConfigService } from "../../services/AppConfigService";
 import { UserService } from "../../services/UserService";
 import { StudentService } from "../../services/StudentService";
 import { StudentRoutesConst } from "../../constants/routes/RouteConstants";
+import { ClientIdType } from "@/types/contracts/ConfigContract";
 
 @Options({
   components: {
     NavBar,
   },
 })
 export default class AppStudent extends Vue {
+  // Flag to hold whole student app children mounding
+  isAuthReady = false;
   async created() {
     const router = this.$router;
+    const route = this.$route;
+    await AppConfigService.shared.initAuthService(ClientIdType.STUDENT);
+    this.isAuthReady = true;
     if (!AppConfigService.shared.authService?.authenticated) {
       router.push({
         name: StudentRoutesConst.LOGIN,
       });
-    } else if (await UserService.shared.checkUser()) {
-      /*After checking the user exists, information differences between BC Service card and SABC is synced
-      And Redirect to Home page (Student's Dashboard)*/
-      await StudentService.shared.synchronizeFromUserInfo();
-      router.push({
-        name: StudentRoutesConst.STUDENT_DASHBOARD,
-      });
     } else {
-      /* User doesn't exist in SABC Database and so redirect the user to Student Profile page
+      // TODO:
+      // - Try to implement a role based processing
+      // Get path
+      if (await UserService.shared.checkUser()) {
+        await StudentService.shared.synchronizeFromUserInfo();
+        if (route.path === "/student") {
```

Comment:
Is there a reason why we need to check if the path contains "/student" here?
This component will be loaded only under "/student" path, right?